android/jni/Android.mk

```diff
@@ -21,40 +21,35 @@ ifeq ($(TARGET_ARCH_ABI), x86_64)
     MY_ARCH_FLAGS := X86_64
 endif
 
-include $(CLEAR_VARS)
-LOCAL_ARM_MODE := $(MY_ARM_MODE)
-LOCAL_MODULE := cpufeatures
-LOCAL_SRC_FILES := $(NDK_ROOT)/sources/android/cpufeatures/cpu-features.c
-LOCAL_CFLAGS := -Wall -Wextra -Werror
-LOCAL_EXPORT_C_INCLUDES := $(NDK_ROOT)/sources/android/cpufeatures
-LOCAL_EXPORT_LDLIBS := -ldl
-LOCAL_ARM_NEON := ${MY_ARM_NEON}
-include $(BUILD_SHARED_LIBRARY)
-
 include $(CLEAR_VARS)
 LOCAL_ARM_MODE := $(MY_ARM_MODE)
 LOCAL_MODULE := mobileffmpeg_abidetect
-LOCAL_SRC_FILES := $(MY_PATH)/mobileffmpeg_abidetect.c
-LOCAL_CFLAGS := -Wall -Wextra -Werror -Wno-unused-parameter -I${LOCAL_PATH}/../../prebuilt/android-$(TARGET_ARCH)/ffmpeg/include -I$(NDK_ROOT)/sources/android/cpufeatures -DMOBILE_FFMPEG_${MY_ARCH_FLAGS}
+LOCAL_SRC_FILES := mobileffmpeg_abidetect.c
+LOCAL_CFLAGS := -Wall -Wextra -Werror -Wno-unused-parameter -DMOBILE_FFMPEG_${MY_ARCH_FLAGS}
 LOCAL_LDLIBS := -llog -lz -landroid
-LOCAL_SHARED_LIBRARIES := cpufeatures
+LOCAL_SHARED_LIBRARIES := cpu_features libavcodec libavutil
```

Comment:
There was some unresolved symbol that could be found in **libavcodec** (in case of LTS build), but I don't remember exactly how it happened. I will check again.

Reply:
Dependencies given here define which libraries should be loaded in `AbiDetect` class. It is important to keep it short and not depend any `libabxx` library.

Reply:
The problem is that `mobileffmpeg_abidetect.c` includes `fftools_ffmpeg.h` which includes quite a few headers from **libavutil** and **libavcodec**. The result requires [this line](https://github.com/tanersener/mobile-ffmpeg/pull/375/commits/de7f3fcef7aaf58497fdc68ee64bed614c249f9b#diff-983cf7060ee7932325a5473d56993529R3), which could be expressed by a cleaner (IMHO)

    LOCAL_SHARED_LIBRARIES := cpu-features libavutil

(`libavcodec` proved to be unnecessary here)

Reply:
`mobileffmpeg_abidetect.c` includes `fftools_ffmpeg.h` to access `FFMPEG_CONFIGURATION`. But we don't need to depend `libavutil` for that. Just having the include is enough. I think you doing this in the latest version.

Reply:
Yes, that's what I do, but I don't like the ugly path that I put there. You don't like a fake dependency even more, so let it be it.